src/EventStore.Core.Tests/Transforms/BitFlip/BitFlipChunkWriteTransform.cs

```diff
@@ -14,20 +16,19 @@ public ChunkDataWriteStream TransformData(ChunkDataWriteStream dataStream) {
 		return _transformedStream;
 	}
 
-	public void CompleteData(int footerSize, int alignmentSize) {
+	public ValueTask CompleteData(int footerSize, int alignmentSize, CancellationToken token = default) {
 		var chunkHeaderAndDataSize = (int)_transformedStream.ChunkFileStream.Position;
 		var alignedSize = GetAlignedSize(chunkHeaderAndDataSize + footerSize, alignmentSize);
 		var paddingSize = alignedSize - chunkHeaderAndDataSize - footerSize;
-		if (paddingSize > 0) {
-			var padding = new byte[paddingSize];
-			_transformedStream.ChunkFileStream.Write(padding);
-			_transformedStream.ChecksumAlgorithm.TransformBlock(padding, 0, padding.Length, null, 0);
-		}
+
+		return paddingSize > 0
+			? _transformedStream.WriteAsync(new byte[paddingSize], token)
```

Comment:
the bits in the padding will flipped as it's going through the transformed stream, but i think that it doesn't really matter here as we're not testing if the padding has only zeroes

Reply:
Correct. That's why I left this code as-is. Padding for production code is zeroed.